## Задание 1

В файле находится выборка покупок пользователей. Покупать можно либо трансляцию матча, либо подписку на прямой эфир.
http://bit.ly/umatask  
id — идентификатор пользователя
created — дата покупки
Новый сезон РПЛ условно считать начавшимся с 01 июля 2019. Старый сезон — до 1 июля 2019.

Ваша задача развернуть данные в pd.DataFrame

In [1]:
# Your code should be written below:
import pandas as pd
data = pd.read_csv('Task 1.csv')
data.head()

created  \
0  2018-09-02 19:12:22.144047   
1  2018-08-30 13:16:12.620303   
2  2018-09-02 19:27:45.044487   
3  2018-09-01 14:50:46.626348   
4  2018-09-01 18:42:08.190768   

                                               title            name  \
0    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
1     Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3        ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
4  Краснодар - Локомотив. Российская Премьер-лига...  Трансляция РПЛ   

                                 id  
0  fa18fdf154aa0defaf386d52f770ff52  
1  0ae165a76e8730674f61bb91c0aa108b  
2  88b3dc521850617dc465eb7936282357  
3  88dd235c1eeb4be7b39637730e54c121  
4  1d6384cc9a6917f2dd1b7df7a9681eb2

## Подсчитайте следующие статистики:
### Q1: Количество уникальных юзеров:
### Ответ: 163988

In [2]:
# Your code should be written below:
print("Колличество уникальных юзеров:",data['id'].nunique())

Колличество уникальных юзеров: 163988


### Q2: Количество уникального контента:
### Ответ : 29

In [3]:
# Your code should be written below
print("Колличество уникального контента:",data['name'].nunique())

Колличество уникального контента: 29


### Q3: Кто наши покупатели:
Кто покупает подписки в новом сезоне — новые пользователи, или у них были и раньше подписки/трансляции?
### Ответ:  В новом сезоне было 70 клиентов, из них 36 клиентов уже делали покупки в прошлом сезоне. В итоге в новом сезоне 51.5 % покупателей - клиенты, уже приобретавшие подписку, 48.5% - новые клиенты.

Выделим 2 группы: Прошлый и Новый сезон

In [4]:
def group(created):
    if created <= "2019-07-30 23:59:59" :
        return 'Прошлый сезон'
    else:
        return 'Новый сезон'

In [5]:
data['group'] = data['created'].apply(group)
data.head()

created  \
0  2018-09-02 19:12:22.144047   
1  2018-08-30 13:16:12.620303   
2  2018-09-02 19:27:45.044487   
3  2018-09-01 14:50:46.626348   
4  2018-09-01 18:42:08.190768   

                                               title            name  \
0    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
1     Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3        ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
4  Краснодар - Локомотив. Российская Премьер-лига...  Трансляция РПЛ   

                                 id          group  
0  fa18fdf154aa0defaf386d52f770ff52  Прошлый сезон  
1  0ae165a76e8730674f61bb91c0aa108b  Прошлый сезон  
2  88b3dc521850617dc465eb7936282357  Прошлый сезон  
3  88dd235c1eeb4be7b39637730e54c121  Прошлый сезон  
4  1d6384cc9a6917f2dd1b7df7a9681eb2  Прошлый сезон

Посчитаем количество покпупок по сезонам

In [6]:
pokupki = data.groupby('group').agg({'id':'nunique'})
pokupki

id
group                
Новый сезон        71
Прошлый сезон  163954

Посмотрим на клиентов, которые купили в Новом сезоне

In [7]:
ns = data.query('group =="Новый сезон"')
ns_group = ns.groupby('id').agg({'id':'count'})
ns_group.columns = [['new_seasons']]
ns_group.head()

new_seasons
id                                          
060e4eada0ace4d8cd1fe93b35db8ddc           1
08d299a59062ecb2b87824d920380ced           1
08ed8c6e8aaea0271a4709245cb8e386           1
0dad632a7a66b59c7c5a12abefbec2a0           1
1887c5bf1f0a9ab55a5f88d4cc3473ea           1

In [8]:
ns_group = ns_group.reset_index()
ns_group

id new_seasons
0   060e4eada0ace4d8cd1fe93b35db8ddc           1
1   08d299a59062ecb2b87824d920380ced           1
2   08ed8c6e8aaea0271a4709245cb8e386           1
3   0dad632a7a66b59c7c5a12abefbec2a0           1
4   1887c5bf1f0a9ab55a5f88d4cc3473ea           1
..                               ...         ...
66  ec8ec3158dd874ce734a316f81b43408           1
67  ec9e7b486e265177c1b05dafa46b83b2           1
68  ecaef1812b28b3bc18cbb8fdf17241a2           1
69  ee5527d836bdc78e16c910276b872e62           1
70  fb475604eb1685ffa39063922101faff           1

[71 rows x 2 columns]

Посмотрим на клиентов, которые купили в Прошлом сезоне

In [9]:
ps = data.query('group =="Прошлый сезон"')
ps_group = ps.groupby('id').agg({'id':'count'})
ps_group.columns = [['past_seasons']]
ps_group.head()

past_seasons
id                                           
0001da523657aa3186db77e58211c3b1            4
00026393e0f79b4adc7354c8be44267d            1
0003abe208a50801a9c5e967f02dc104            3
000408388975a1257b954e7609952cf1            1
0004965a162053108cda6803ed676532            4

In [10]:
ps_group = ps_group.reset_index()

In [11]:
ps_group.columns=[['id','past_seasons']]

In [12]:
ps_group.head()

id past_seasons
0  0001da523657aa3186db77e58211c3b1            4
1  00026393e0f79b4adc7354c8be44267d            1
2  0003abe208a50801a9c5e967f02dc104            3
3  000408388975a1257b954e7609952cf1            1
4  0004965a162053108cda6803ed676532            4

In [13]:
itog = ns_group.merge(ps_group)
itog

id new_seasons past_seasons
0   060e4eada0ace4d8cd1fe93b35db8ddc           1            5
1   08d299a59062ecb2b87824d920380ced           1            2
2   1a3507700018ddea3e369ab32106be00           1            2
3   1ed973dbbea186015bc410530135f5f1           1            3
4   2f4e448d456df403d70a9078c71d9fff           1            1
5   35ca8cebec2cdcf2667174b6d131b1d2           1            3
6   3690a0c1be3a4b6105fee6bba1e11b1b           1            1
7   3b1a9d491623b9f41f9cc65854295f40           1           28
8   3d070f1c4703e72d90ee90c7ab444202           1           17
9   50ee40a23dfb7208aa664b21219cdcf0           1            8
10  53d38b3e49ab0a699d0fbc6223e72577           1            4
11  6902712b123e80ef2bd776481e42d86c           1            4
12  7ebc941e23ede1634b00843e6ce6e292           1           11
13  89c97d13d6a592b5ea70e9315726e277           1            4
14  8ea734f1a1d4f84fbb7dd04d465b8aad           1            5
15  9525a8c69de1caf77dcf6450b42c911d           1           22
16  99771eefc5a62da2919441642615e587           1            4
17  a0836bf6ef8d8f77af2ccca4aa19bae8           1            1
18  a308ce0602663cf62872e371b6587c1c           1            7
19  b63a1712677b109bb1e25e1bc045fbc5           1            4
20  b99e412b0bcad169b4fd316248e30ec9           1            1
21  bf326621624c04ea73b9dcb09c0c2358           1            3
22  c2a6ae86aa5028ba3927d9d8aaf21dce           1           13
23  c564ed25510b7f33ae6781d1418e8114           1            1
24  cb4411153592d9de4a5dedb0f41d2430           1            1
25  d4618885a4e3ea258dfe461f407e9943           2            2
26  d6f0ee79735acca19134b6a0c55eadb2           1            9
27  da0af46246f61e9c5136e37c990bbf65           1            1
28  e0efa08e66cecd844833573967fbd55f           1           12
29  e277270a83b717cb3de02a4cbd68245f           1            6
30  e838e2774e8f9a9d20c69f828a3dc75c           1            1
31  eb527e879bc78b76b30f9ab91d49cb6e           1            3
32  ec8ec3158dd874ce734a316f81b43408           1            2
33  ec9e7b486e265177c1b05dafa46b83b2           1            1
34  ecaef1812b28b3bc18cbb8fdf17241a2           1            2
35  ee5527d836bdc78e16c910276b872e62           1            4
36  fb475604eb1685ffa39063922101faff           1            1

### В новом сезоне было 70 клиентов, из них 36 клиентов уже делали покупки в прошлом сезоне. В итоге в новом сезоне 51.5 % покупателей - клиенты, уже приобретавшие подписку, 48.5% - новые клиенты.

### Q5: Отдельно рассмотрим тех, у кого в прошлом сезоне было куплено более одной подписки. Хочу прокоммуницировать с этими ребятами.
Что с ними в этом сезоне? Они подписались? Стали покупать разовые? Ничего не купили ещё?
### Ответ: В новом году подписались 26 клиентов из прошлого сезона, чаще всего покупали Матч Премьер и Трансляции РПЛ

Выведем клиентов, которые купили в прошлом сезоне больше 1 раза

In [14]:
# Your code should be written below
ps1 = data.query('group =="Прошлый сезон"')
ps_group1 = ps.groupby('id').agg({'id':'count'})
ps_group1 = ps_group1.query('id > 1')
list_ps = ps_group1.index.to_list()

In [15]:
data_ps_1 = data[data['id'].isin(list_ps)]
data_ps_1

created  \
1       2018-08-30 13:16:12.620303   
2       2018-09-02 19:27:45.044487   
3       2018-09-01 14:50:46.626348   
5       2018-09-23 16:33:15.972475   
7       2018-09-01 16:32:56.004800   
...                            ...   
410297  2018-08-13 20:18:48.670870   
410298  2018-08-19 18:20:47.825524   
410299  2018-08-16 20:14:47.946353   
410300  2018-08-11 21:34:31.617997   
410301  2018-08-26 16:39:02.069440   

                                                    title            name  \
1          Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2         Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3             ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
5                         Прямой эфир канала МАТЧ ПРЕМЬЕР    МАТЧ ПРЕМЬЕР   
7             ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
...                                                   ...             ...   
410297      Рубин - Зенит. Российская Премьер-Лига. 3 тур  Трансляция РПЛ   
410298       Зенит - Урал. Российская Премьер-лига. 4 тур  Трансляция РПЛ   
410299  Лига Европы. 3-й отборочный раунд. Зенит - Дин...  Трансляция РПЛ   
410300  Ростов - Крылья Советов. Российская Премьер-Ли...  Трансляция РПЛ   
410301        Уфа - Зенит. Российская Премьер-лига. 5 тур  Трансляция РПЛ   

                                      id          group  
1       0ae165a76e8730674f61bb91c0aa108b  Прошлый сезон  
2       88b3dc521850617dc465eb7936282357  Прошлый сезон  
3       88dd235c1eeb4be7b39637730e54c121  Прошлый сезон  
5       674a8d048b63bc3d4abd8f416206a83e  Прошлый сезон  
7       7b707be012ee24ac716aad7caadbc9db  Прошлый сезон  
...                                  ...            ...  
410297  3d6c20b1e040455755d52705ca658dd3  Прошлый сезон  
410298  c4252752998f20fbc6177890ba6eb2fc  Прошлый сезон  
410299  3710ba63b209341c112ad7dc430995ca  Прошлый сезон  
410300  8882b6c4efae68e16f47eee41d76594a  Прошлый сезон  
410301  2db915f90cb0473bcc28a5abe6cb99cd  Прошлый сезон  

[316919 rows x 5 columns]

In [16]:
print('Количество клиентов, купивших в прошлом сезоне:',data_ps_1['id'].nunique())

Количество клиентов, купивших в прошлом сезоне: 70616


Самые популярные покупки среди клиентов в прошлом сезоне

In [17]:
name_ps_1 = data_ps_1.groupby('name').agg({'id':'nunique'}).sort_values('id',ascending=False)
name_ps_1

id
name                                                   
Трансляция РПЛ                                    59528
МАТЧ ПРЕМЬЕР                                      21370
РПЛ на championat.com                              4060
РПЛ на sport24.ru                                  1182
РПЛ на sport-express.ru                             507
РПЛ на soccer365.ru                                 251
МАТЧ ПРЕМЬЕР на premierliga.ru                      205
«Краснодар» (Краснодар) в сезоне РПЛ 2018/2019      181
«Крылья Советов» (Самара) в сезоне РПЛ 2018/2019    136
РПЛ на sportmail.ru                                  80
«Краснодар» в сезоне РПЛ 19/20                       54
«Оренбург» (Оренбург) в сезоне РПЛ 2018/2019         41
«Крылья Советов» в сезоне РПЛ 19/20                  36
«Урал» (Екатеринбург) в сезоне РПЛ 2018/2019         34
«Арсенал» (Тула) в сезоне РПЛ 2018/2019              27
РПЛ на nevasport.ru                                   8
«Спартак» (Москва) в сезоне РПЛ 2018/2019             8
РПЛ на Рамблер                                        8
«Оренбург» в сезоне РПЛ 19/20                         7
«Арсенал» в сезоне РПЛ 19/20                          6
«Уфа» в сезоне РПЛ 2018/2019                          6
«Зенит» (Санкт-Петербург) в сезоне РПЛ 2018/2019      5
РПЛ 18/19 (на сайте РПЛ)                              4
РПЛ 19/20 (на сайте РПЛ)                              4
МАТЧ ПРЕМЬЕР на soccer365.ru                          1
МАТЧ ПРЕМЬЕР «Крылья Советов»                         1
РПЛ на sportsdaily.ru                                 1
МАТЧ ПРЕМЬЕР на nevasport.ru                          1

Что купили клиенты прошлого сезона в новом сезоне:

In [18]:
ps_noviy = data.query('group == "Новый сезон"')

In [19]:
late_client=data_ps_1.merge(ps_noviy)
late_client

created  \
0   2019-08-01 21:55:39.865035   
1   2019-08-01 10:54:09.117805   
2   2019-08-01 15:26:14.523003   
3   2019-07-31 09:46:24.914273   
4   2019-08-01 12:29:43.999826   
5   2019-08-01 22:41:35.023107   
6   2019-08-01 15:14:59.156926   
7   2019-08-01 14:44:22.305276   
8   2019-08-01 18:11:29.845358   
9   2019-08-01 19:13:01.381181   
10  2019-07-31 19:30:11.491477   
11  2019-08-01 17:17:23.433895   
12  2019-07-31 14:26:54.550908   
13  2019-08-01 14:42:51.013036   
14  2019-08-01 20:08:34.409840   
15  2019-08-01 11:55:13.882692   
16  2019-08-01 22:42:54.999819   
17  2019-07-31 12:34:22.652213   
18  2019-08-01 17:00:28.293674   
19  2019-08-01 17:13:12.325242   
20  2019-08-01 00:03:20.553192   
21  2019-08-01 21:11:02.989773   
22  2019-08-01 21:45:38.657135   
23  2019-08-01 19:35:41.254825   
24  2019-08-01 20:47:52.447537   
25  2019-08-01 20:00:23.423773   
26  2019-08-01 11:50:33.687943   
27  2019-08-01 16:01:49.860456   

                                                title  \
0                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
1                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
2                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
3                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
4                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
5                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
6                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
7                     Прямой эфир канала МАТЧ ПРЕМЬЕР   
8    Спартак - Динамо. Российская Премьер-лига. 4 тур   
9      Зенит - Тамбов. Российская Премьер-лига. 1 тур   
10   Оренбург - Зенит. Российская Премьер-лига. 3 тур   
11  Зенит - Краснодар. Российская Премьер-лига. 4 тур   
12   Спартак - Динамо. Российская Премьер-лига. 4 тур   
13       Рубин - ЦСКА. Российская Премьер-лига. 4 тур   
14   Краснодар - Сочи. Российская Премьер-лига. 3 тур   
15                    Прямой эфир канала МАТЧ ПРЕМЬЕР   
16                    Прямой эфир канала МАТЧ ПРЕМЬЕР   
17                    Прямой эфир канала МАТЧ ПРЕМЬЕР   
18  Крылья Советов - Локомотив. Российская Премьер...   
19  Крылья Советов - Локомотив. Российская Премьер...   
20   Спартак - Динамо. Российская Премьер-лига. 4 тур   
21   Спартак - Динамо. Российская Премьер-лига. 4 тур   
22                    Прямой эфир канала МАТЧ ПРЕМЬЕР   
23                    Прямой эфир канала МАТЧ ПРЕМЬЕР   
24      Урал - Ростов. Российская Премьер-лига. 4 тур   
25       Рубин - ЦСКА. Российская Премьер-лига. 4 тур   
26   Спартак - Динамо. Российская Премьер-лига. 4 тур   
27                    Прямой эфир канала МАТЧ ПРЕМЬЕР   

                              name                                id  \
0                     МАТЧ ПРЕМЬЕР  060e4eada0ace4d8cd1fe93b35db8ddc   
1                     МАТЧ ПРЕМЬЕР  1ed973dbbea186015bc410530135f5f1   
2                     МАТЧ ПРЕМЬЕР  53d38b3e49ab0a699d0fbc6223e72577   
3                     МАТЧ ПРЕМЬЕР  08d299a59062ecb2b87824d920380ced   
4   МАТЧ ПРЕМЬЕР на premierliga.ru  99771eefc5a62da2919441642615e587   
5                     МАТЧ ПРЕМЬЕР  6902712b123e80ef2bd776481e42d86c   
6                     МАТЧ ПРЕМЬЕР  8ea734f1a1d4f84fbb7dd04d465b8aad   
7                     МАТЧ ПРЕМЬЕР  eb527e879bc78b76b30f9ab91d49cb6e   
8                   Трансляция РПЛ  c2a6ae86aa5028ba3927d9d8aaf21dce   
9                   Трансляция РПЛ  1a3507700018ddea3e369ab32106be00   
10                  Трансляция РПЛ  a308ce0602663cf62872e371b6587c1c   
11                  Трансляция РПЛ  ecaef1812b28b3bc18cbb8fdf17241a2   
12                  Трансляция РПЛ  b63a1712677b109bb1e25e1bc045fbc5   
13                  Трансляция РПЛ  d6f0ee79735acca19134b6a0c55eadb2   
14                  Трансляция РПЛ  e0efa08e66cecd844833573967fbd55f   
15                    МАТЧ ПРЕМЬЕР  e277270a83b717cb3de02a4cbd68245f   
16                    МАТЧ ПРЕМЬЕР  ee5527d836bdc78e16c910276b872e62   
17                    МАТЧ ПРЕМЬЕР  7ebc941e23ede1634b00843e6ce6e292   
18            

In [20]:
late_client_group= late_client.groupby('name').agg({'id':'nunique'}).sort_values('id',ascending=False)
late_client_group

id
name                              
МАТЧ ПРЕМЬЕР                    13
Трансляция РПЛ                  13
МАТЧ ПРЕМЬЕР на premierliga.ru   1

В новом году подписались 26 клиентов из прошлого сезона, чаще всего покупали Матч Премьер и Трансляции РПЛ

### Q6: Отдельно рассмотрим тех, у кого в прошлом сезоне было куплено более одной подписки. Хочу прокоммуницировать с этими ребятами.
Может, уже есть те, кто покупал трансляции 2 и более раз? Им выгоднее подписка. Тоже думаем сделать под них специальный оффер и прокоммуницировать. Сколько таких?
### Ответ: 70616, чаще всего покупали Трансляции РПЛ и Матч Премьер

In [21]:
ps2 = data.query('group =="Прошлый сезон"')
ps_group2 = ps.groupby('id').agg({'id':'count'})
ps_group2 = ps_group2.query('id >= 2')
list_ps_1 = ps_group2.index.to_list()

In [22]:
data_ps_2 = data[data['id'].isin(list_ps_1)]
data_ps_2

created  \
1       2018-08-30 13:16:12.620303   
2       2018-09-02 19:27:45.044487   
3       2018-09-01 14:50:46.626348   
5       2018-09-23 16:33:15.972475   
7       2018-09-01 16:32:56.004800   
...                            ...   
410297  2018-08-13 20:18:48.670870   
410298  2018-08-19 18:20:47.825524   
410299  2018-08-16 20:14:47.946353   
410300  2018-08-11 21:34:31.617997   
410301  2018-08-26 16:39:02.069440   

                                                    title            name  \
1          Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2         Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3             ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
5                         Прямой эфир канала МАТЧ ПРЕМЬЕР    МАТЧ ПРЕМЬЕР   
7             ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
...                                                   ...             ...   
410297      Рубин - Зенит. Российская Премьер-Лига. 3 тур  Трансляция РПЛ   
410298       Зенит - Урал. Российская Премьер-лига. 4 тур  Трансляция РПЛ   
410299  Лига Европы. 3-й отборочный раунд. Зенит - Дин...  Трансляция РПЛ   
410300  Ростов - Крылья Советов. Российская Премьер-Ли...  Трансляция РПЛ   
410301        Уфа - Зенит. Российская Премьер-лига. 5 тур  Трансляция РПЛ   

                                      id          group  
1       0ae165a76e8730674f61bb91c0aa108b  Прошлый сезон  
2       88b3dc521850617dc465eb7936282357  Прошлый сезон  
3       88dd235c1eeb4be7b39637730e54c121  Прошлый сезон  
5       674a8d048b63bc3d4abd8f416206a83e  Прошлый сезон  
7       7b707be012ee24ac716aad7caadbc9db  Прошлый сезон  
...                                  ...            ...  
410297  3d6c20b1e040455755d52705ca658dd3  Прошлый сезон  
410298  c4252752998f20fbc6177890ba6eb2fc  Прошлый сезон  
410299  3710ba63b209341c112ad7dc430995ca  Прошлый сезон  
410300  8882b6c4efae68e16f47eee41d76594a  Прошлый сезон  
410301  2db915f90cb0473bcc28a5abe6cb99cd  Прошлый сезон  

[316919 rows x 5 columns]

In [23]:
data_ps_2['id'].nunique()

70616

In [24]:
name_ps_2 = data_ps_2.groupby('name').agg({'id':'nunique'}).sort_values('id',ascending=False)
name_ps_2

id
name                                                   
Трансляция РПЛ                                    59528
МАТЧ ПРЕМЬЕР                                      21370
РПЛ на championat.com                              4060
РПЛ на sport24.ru                                  1182
РПЛ на sport-express.ru                             507
РПЛ на soccer365.ru                                 251
МАТЧ ПРЕМЬЕР на premierliga.ru                      205
«Краснодар» (Краснодар) в сезоне РПЛ 2018/2019      181
«Крылья Советов» (Самара) в сезоне РПЛ 2018/2019    136
РПЛ на sportmail.ru                                  80
«Краснодар» в сезоне РПЛ 19/20                       54
«Оренбург» (Оренбург) в сезоне РПЛ 2018/2019         41
«Крылья Советов» в сезоне РПЛ 19/20                  36
«Урал» (Екатеринбург) в сезоне РПЛ 2018/2019         34
«Арсенал» (Тула) в сезоне РПЛ 2018/2019              27
РПЛ на nevasport.ru                                   8
«Спартак» (Москва) в сезоне РПЛ 2018/2019             8
РПЛ на Рамблер                                        8
«Оренбург» в сезоне РПЛ 19/20                         7
«Арсенал» в сезоне РПЛ 19/20                          6
«Уфа» в сезоне РПЛ 2018/2019                          6
«Зенит» (Санкт-Петербург) в сезоне РПЛ 2018/2019      5
РПЛ 18/19 (на сайте РПЛ)                              4
РПЛ 19/20 (на сайте РПЛ)                              4
МАТЧ ПРЕМЬЕР на soccer365.ru                          1
МАТЧ ПРЕМЬЕР «Крылья Советов»                         1
РПЛ на sportsdaily.ru                                 1
МАТЧ ПРЕМЬЕР на nevasport.ru                          1

### Q7: А матчи с какими командами покупали больше всего?
### Ответ: из категории выбирали чаще всего Краснодар, а самая популярная трансляция - Спартак-ЦСКА

In [25]:
# Your code should be written below
data['comand'] = data['name'].str.extract('(«.*») ', expand=False)
data.head()                            

created  \
0  2018-09-02 19:12:22.144047   
1  2018-08-30 13:16:12.620303   
2  2018-09-02 19:27:45.044487   
3  2018-09-01 14:50:46.626348   
4  2018-09-01 18:42:08.190768   

                                               title            name  \
0    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
1     Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2    Зенит – Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3        ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
4  Краснодар - Локомотив. Российская Премьер-лига...  Трансляция РПЛ   

                                 id          group comand  
0  fa18fdf154aa0defaf386d52f770ff52  Прошлый сезон    NaN  
1  0ae165a76e8730674f61bb91c0aa108b  Прошлый сезон    NaN  
2  88b3dc521850617dc465eb7936282357  Прошлый сезон    NaN  
3  88dd235c1eeb4be7b39637730e54c121  Прошлый сезон    NaN  
4  1d6384cc9a6917f2dd1b7df7a9681eb2  Прошлый сезон    NaN

In [26]:
data['comand'].unique()

array([nan, '«Оренбург»', '«Спартак»', '«Краснодар»', '«Арсенал»',
       '«Урал»', '«Крылья Советов»', '«Уфа»', '«Зенит»'], dtype=object)

In [27]:
comand_group = data.groupby('comand').agg({'id':'count'}).sort_values('id',ascending=False)
comand_group

id
comand               
«Краснодар»       679
«Крылья Советов»  473
«Оренбург»        159
«Арсенал»          99
«Урал»             88
«Уфа»              19
«Спартак»          13
«Зенит»             7

Посмотрим на самые популярные матчи

In [28]:
data['title'] = data['title'].str.replace('–','-')

In [29]:
data['comand2'] = data['title'].str.extract('(.*-.*.-)', expand=False)
data.head()  

created  \
0  2018-09-02 19:12:22.144047   
1  2018-08-30 13:16:12.620303   
2  2018-09-02 19:27:45.044487   
3  2018-09-01 14:50:46.626348   
4  2018-09-01 18:42:08.190768   

                                               title            name  \
0    Зенит - Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
1     Ростов - Рубин. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
2    Зенит - Спартак. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
3        ЦСКА - Урал. Российская Премьер-лига. 6 тур  Трансляция РПЛ   
4  Краснодар - Локомотив. Российская Премьер-лига...  Трансляция РПЛ   

                                 id          group comand  \
0  fa18fdf154aa0defaf386d52f770ff52  Прошлый сезон    NaN   
1  0ae165a76e8730674f61bb91c0aa108b  Прошлый сезон    NaN   
2  88b3dc521850617dc465eb7936282357  Прошлый сезон    NaN   
3  88dd235c1eeb4be7b39637730e54c121  Прошлый сезон    NaN   
4  1d6384cc9a6917f2dd1b7df7a9681eb2  Прошлый сезон    NaN   

                                      comand2  
0        Зенит - Спартак. Российская Премьер-  
1         Ростов - Рубин. Российская Премьер-  
2        Зенит - Спартак. Российская Премьер-  
3            ЦСКА - Урал. Российская Премьер-  
4  Краснодар - Локомотив. Российская Премьер-

In [30]:
comand_group2 = data.groupby('comand2').agg({'id':'count'}).sort_values('id',ascending=False)
comand_group2

id
comand2                                           
Спартак - ЦСКА. Российская Премьер-          29305
Ростов - Спартак. Российская Премьер-        16863
Спартак - Зенит. Российская Премьер-         11809
ЦСКА - Зенит. Российская Премьер-            10211
Спартак - Динамо. Российская Премьер-         8263
...                                            ...
Урал - Анжи. Российская Премьер-                 3
Ахмат - Енисей. Российская Премьер-              3
ИНТЕРШУМ Динамо - Урал. Российская Премьер-      2
Уфа - Анжи. Российская Премьер-                  2
Урал - Динамо Москва. Российская Премьер-        1

[248 rows x 1 columns]

## Задание 2

Вы провели А/Б-тест двух посадочных страниц. На вариант А зашли 6 352 человека, 462 из них сконвертировались. На вариант Б зашли 7 123 человека, это дало 556 конверсий. Можно ли по результатам теста определить, какой вариант лучше?
### Ответ : нет

In [31]:
from scipy import stats as st
import numpy as np
import math as mth

alpha = .05 # критический уровень статистической значимости

successes = np.array([462, 556])
trials = np.array([6352, 7123])

# пропорция успехов в первой группе:
p1 = successes[0]/trials[0]

# пропорция успехов во второй группе:
p2 = successes[1]/trials[1]

# пропорция успехов 
p_combined = (successes[0] + successes[1]) / (trials[0] + trials[1])

# разница пропорций 
difference = p1 - p2 

# считаем статистику в ст.отклонениях стандартного нормального распределения
z_value = difference / mth.sqrt(p_combined * (1 - p_combined) * (1/trials[0] + 1/trials[1]))

# задаем стандартное нормальное распределение (среднее 0, ст.отклонение 1)
distr = st.norm(0, 1) 

p_value = (1 - distr.cdf(abs(z_value))) * 2

print('p-значение: ', p_value)

if (p_value < alpha):
    print("Отвергаем нулевую гипотезу: между вариантами есть значимая разница")
else:
    print("Не получилось отвергнуть нулевую гипотезу, нет оснований считать варианты разными") 

p-значение:  0.24306089342860648
Не получилось отвергнуть нулевую гипотезу, нет оснований считать варианты разными


## Задание 3

Сколько людей должно принять участие в эксперименте из предыдущего вопроса, чтобы мы смогли однозначно определить победителя со статистической значимостью 5% и мощностью 80%. Будем считать, что вариант Б победил, если его конверсия хотя бы на 10% выше относительно варианта А.

In [34]:
19623

19623

## Задание 4
Если вам надо быстро сравнить много продуктов из схожей категории и вы можете выбрать только две метрики для сравнения, какие вы возьмете?

средний чек и прибыль